### PR TITLE
rebase 9.19, rm conference

### PR DESCRIPTION
**Pull request by greenmemo**
rtc2, rebase 7.18, squash

### CODE REVIEW

File: local.properties [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## This file is automatically generated by Android Studio.
+# Do not modify this file -- YOUR CHANGES WILL BE ERASED!
+#
+# This file must *NOT* be checked into Version Control Systems,
+# as it contains information specific to your local configuration.
+#
+# Location of the SDK. This is only used by Gradle.
+# For customization when using a Version Control System, please read the
+# header note.
+#Thu May 26 19:53:02 CST 2016
```
Comment (githubzhaoliang) [inline local.properties]: key store should not release to demo

File: src/com/hyphenate/chatuidemo/DemoHelper.java [hunk 2]
```diff
@@ -755,6 +755,11 @@ public void onCmdMessageReceived(List<EMMessage> messages) {
                             broadcastManager.sendBroadcast(new Intent(RedPacketConstant.REFRESH_GROUP_RED_PACKET_ACTION));
                         }
                     }
+
+                    if (action.equals("__Call_ReqP2P_ConferencePattern")) {
```
Comment (githubzhaoliang) [inline src/com/hyphenate/chatuidemo/DemoHelper.java]: test code？

File: src/com/hyphenate/chatuidemo/ui/CallActivity.java [hunk 3]
```diff
@@ -58,6 +68,62 @@
...
         super.onCreate(arg0);
         audioManager = (AudioManager) this.getSystemService(Context.AUDIO_SERVICE);
+        pushProvider = new EMCallManager.EMCallPushProvider() {
+            
+            void updateMessageText(final EMMessage oldMsg, final String to) {
+                // update local message text
+                EMConversation conv = EMClient.getInstance().chatManager().getConversation(oldMsg.getTo());
+                conv.removeMessage(oldMsg.getMsgId());
+                
+                EMMessage newMsg = EMMessage.createTxtSendMessage("Opposite is offline", to);
```
Comment (githubzhaoliang) [inline src/com/hyphenate/chatuidemo/ui/CallActivity.java]: what is this message for?

File: src/com/hyphenate/chatuidemo/ui/CallActivity.java [hunk 4]
```diff
@@ -58,6 +68,62 @@
...
+            }
+            
+            @Override
+            public void onSendPushMessage(final String to) {
+
+                //this function should exposed & move to Demo
+                EMLog.d(TAG, "onSendPushMessage, to:" + to);
+                
+                final EMMessage message = EMMessage.createTxtSendMessage("You have an incoming call", to);         
+                // set the user-defined extension field
```
Comment (githubzhaoliang) [inline src/com/hyphenate/chatuidemo/ui/CallActivity.java]: i think this send message operation should be inside SDK.

File: src/com/hyphenate/chatuidemo/ui/CallActivity.java [hunk 5]
```diff
@@ -58,6 +68,62 @@
...
+                message.setAttribute("em_apns_ext", true);
```
Comment (githubzhaoliang) [inline src/com/hyphenate/chatuidemo/ui/CallActivity.java]: 原来这个字段后面是一个json object？

message.setAttribute("em_apns_ext",(new JSONObject()).put("em_push_title","自定义信息"));

File: src/com/hyphenate/chatuidemo/ui/VideoCallActivity.java [hunk 6]
```diff
@@ -113,7 +115,14 @@ protected void onCreate(Bundle savedInstanceState) {
...
         setContentView(R.layout.em_activity_video_call);
-        
+
+        /**
+         * This function is only meaningful when your app need recording feature
+         * If not, remove it.
+         * This function need be called before the video stream started, so we set it in onCreate function.
+         * This method will set the preferred video record encoding codec.
+         * Using default encoding format, recorded file can not be played by mobile player.
+         */
```
Comment (githubzhaoliang) [inline src/com/hyphenate/chatuidemo/ui/VideoCallActivity.java]: 注释说function，但下面一句只是一个变量赋值？

File: src/com/hyphenate/chatuidemo/ui/VoiceCallActivity.java [hunk 7]
```diff
@@ -121,6 +122,9 @@ protected void onCreate(Bundle savedInstanceState) {
 			ringtone = RingtoneManager.getRingtone(this, ringUri);
 			ringtone.play();
 		}
+        final int MAKE_CALL_TIMEOUT = 50 * 1000;
```
Comment (githubzhaoliang) [inline src/com/hyphenate/chatuidemo/ui/VoiceCallActivity.java]: sdk内部应该有一个超时的值

File: src/com/hyphenate/chatuidemo/DemoHelper.java (hunk 2 above)
Comment (greenmemo) [inline src/com/hyphenate/chatuidemo/DemoHelper.java]: when release, remove

**Pull request closed without merging**